src/test/java/com/rtbhouse/kafka/workers/integration/ShutdownTest.java

```diff
@@ -124,17 +126,13 @@ public void shouldShutDownWhenThreadsCannotBeInterrupted() throws InterruptedExc
 
         //then
         assertThat(kafkaWorkers.getStatus()).isEqualTo(CANNOT_STOP_THREADS);
-        assertThat(shutdownTime).isBetween(
```

Comment:
To się jakoś wyprostowało w nowej wersji czy nie było sensu tego sprawdzać?

Reply:
2.4.1 client unregisters partitions on closing which triggers KafkaWorkersImpl.unregister -> TaskManager.unregister -> stopThreads. Threads cannot be stopped, so it waits till a timeout on record processing. While using 2.3.0 client consumer-thread was not blocked in this manner.
That's why I removed checking shutdown time in this test.